src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -122,6 +128,28 @@ void fillInnerParts() {
         commandBoxPlaceholder.getChildren().add(commandBox.getRoot());
     }
 
+
+    /**
+     * Switches the children inside the holder with the tab
+     * @param holder container for tab
+     * @param tab specified tab to be swapped in
+     */
+    public void switchTab(StackPane holder, UiPart<Region> tab) {
+        logger.info("Switching tab");
+        holder.getChildren().clear();
+        holder.getChildren().add(tab.getRoot());
+    }
+
+    /**
+     * Switches the children isnide of viewFinder with the specified tab
+     * @param tab specified tab to be swapped in
+     */
+    public void switchTab(UiPart<Region> tab) {
+        logger.info("Switching tab");
+        viewFinderPlaceholder.getChildren().clear();
+        viewFinderPlaceholder.getChildren().add(tab.getRoot());
```

Comment:
Again, a minor point, its just about reducing overhead when you're reading these names / function definition, as well as reducing duplication of code.

If you name the second one `switchTab(UiPart<Region> tab)`, I recognise you're making an overloaded method that defaults the holder to `this.viewFinderPlaceholder`. Could we make that immediately obvious and reduce code duplication by just calling it like that?

```suggestion
        switchTab(this.viewFinderPlaceholder, tab);
```